### DISCUSSION TITLE
How to ensure unique node scheduling

### DISCUSSION THREAD

**Discussion by muchengyang** [apache/dolphinscheduler#12464] [Q&A]
When the node changes, how to ensure that the task at the same time will only be scheduled by one master node, I don’t see the only guarantee when the instance is generated, how can the following code guarantee it, I don’t understand, please enlighten me

![image](https://user-images.githubusercontent.com/16506338/196907963-28148857-0453-4562-a437-affcb3a2e8a4.png)

![image](https://user-images.githubusercontent.com/16506338/196908038-65b08431-2e8a-4206-961e-f5ee73b1f9e3.png)

**Comment by muchengyang** [comment]
Generated instances are not unique

**Comment by DarkAssassinator** [comment]
Because ds use database transaction
